### PR TITLE
Adding methods to create EncryptedChoice and Ciphertext

### PR DESCRIPTION
**Pull request by svylabs**
We are using this library on one of our applications and we need a way to serialize and deserialize EncryptedChoice and Ciphertext, while the serialization and deserialization can be performed outside of the library, we need the feature to reconstruct these structures again in this library.

### CODE REVIEW

File: src/app/choice.rs [hunk 1]
```diff
@@ -406,8 +406,21 @@ impl<G: Group, S: ProveSum<G>> EncryptedChoice<G, S> {
     pub fn sum_proof(&self) -> &S::Proof {
         &self.sum_proof
     }
+
+
+    /// Returns a new encrypted choice
+    pub fn new_encrypted_choice(choices: Vec<Ciphertext<G>>, range_proof: RingProof<G>, sum_proof: S::Proof) -> Self {
```
Comment (slowli) [inline src/app/choice.rs:412]: Bikeshedding: I'd prefer that this method would be called something like `from_parts`. Also, the docs should specify that if parts do not represent a valid encrypted choice, it cannot be used successfully (i.e., the caller is responsible for the args validation).

File: src/encryption.rs [hunk 2]
```diff
@@ -151,6 +151,14 @@ impl<G: Group> Ciphertext<G> {
         G::serialize_element(&self.blinded_element, &mut bytes[G::ELEMENT_SIZE..]);
         bytes
     }
+
+    pub fn new(random_element: G::Element, blinded_element: G::Element) -> Self {
```
Comment (slowli) [inline src/encryption.rs:155]: Again, `from_parts` looks like a better name for this method. Also, it should have docs.

**Review by slowli — COMMENTED** [review]
Just to understand your use case better: do I understand correctly that using `serde` for (de)serialization is not an option? Both `Ciphertext` and and `EncryptedChoice` implement `Serialize` / `Deserialize` if the crate feature `serde` is enabled.

**Comment by svylabs** [comment]
@slowli  thanks for the suggestion. I checked the code again, you are right, these structs already derives  Deserialize/Serialize. I'm trying to integrate that and will let you know if this pull request is still needed or not.  If that works - I'll close this request.

**Comment by svylabs** [comment]
@slowli  - I tried to use the serde_json to serialize and deserialize. It's working fine when I run unit tests on my application, but I'm unable to compile the code as cosmwasm contract because of the use of `usize` within the mentioned structs: https://medium.com/cosmwasm/debugging-floating-point-generation-in-rust-wasm-smart-contract-f47d833b5fba

So we may need a custom serializer and deserializer if the application will be compiled to cosmwasm contract. For now, I'll just use the fork of your library to see if I encounter any other issues. I'll close the pull request.

**Comment by svylabs** [comment]
Closing this as per previous comment.

**Pull request closed without merging**